test/e2e/service_latency.go

```diff
@@ -144,58 +165,155 @@ func runServiceLatencies(f *Framework, inParallel, total int) (output []time.Dur
 	return output, nil
 }
 
-func singleServiceLatency(f *Framework, i int) (time.Duration, error) {
-	// Make an RC with a single pod.
-	cfg := RCConfig{
-		Client:       f.Client,
-		Image:        "gcr.io/google_containers/pause:1.0",
-		Name:         fmt.Sprintf("trial-%v", i),
-		Namespace:    f.Namespace.Name,
-		Replicas:     1,
-		PollInterval: time.Second,
+type endpointQuery struct {
```

Comment:
Old code, but I'm pretty sure that you need to handle closed channels in the above for loop.  Github wouldn't let me put this comment there, so it's here :-)